modules/fx/arrow-fx/src/main/kotlin/arrow/fx/typeclasses/Bracket.kt

```diff
@@ -145,5 +145,5 @@ interface Bracket<F, E> : MonadError<F, E> {
    *
    */
   fun <A> Kind<F, A>.guaranteeCase(finalizer: (ExitCase<E>) -> Kind<F, Unit>): Kind<F, A> =
-    bracketCase({ _, e -> finalizer(e) }, { this })
+    just<Unit>(Unit).bracketCase({ _, e -> finalizer(e) }, { this })
```

Comment:
Just me trying to understand what's going on. (We just had a training and I'm beginning to grasp some things). The chaining of `just<Unit>(Unit)` first is meant to trigger a boundary?

Reply:
If you remember we discussed the difference between `bracket` and `guarantee` is that `bracketCase` makes the `acquire` step uncancelable.

So you should use `guarantee` to schedule finalizers for your (cancelable) program, while you use `bracket` to guarantee that your `release` step is always called for your acquired resource. Doing so involves an uncancelable boundary between `acquire` and `use`.

In this case, we derive `guarantee` by providing a `Unit` resource, running our program in `use` which ensures that a finalizer is called for a `Unit` resource.